buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/transform/warnings/InjectAllowedWarnings.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.test.rest.transform.warnings;
+
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.elasticsearch.gradle.test.rest.transform.RestTestTransformByParentObject;
+import org.elasticsearch.gradle.test.rest.transform.feature.FeatureInjector;
+import org.gradle.api.tasks.Input;
+
+import java.util.List;
+
+/**
+ * A transformation to inject an allowed warning.
+ */
+public class InjectAllowedWarnings extends FeatureInjector implements RestTestTransformByParentObject {
+
+    private static JsonNodeFactory jsonNodeFactory = JsonNodeFactory.withExactBigDecimals(false);
+
+    private final List<String> allowedWarnings;
+
+    /**
+     * @param allowedWarnings The allowed warnings to inject
+     */
+    public InjectAllowedWarnings(List<String> allowedWarnings) {
+        this.allowedWarnings = allowedWarnings;
+    }
+
+    @Override
+    public void transformTest(ObjectNode doNodeParent) {
+        ObjectNode doNodeValue = (ObjectNode) doNodeParent.get(getKeyToFind());
+        ArrayNode arrayWarnings = (ArrayNode) doNodeValue.get("allowed_warnings");
+        if (arrayWarnings == null) {
+            arrayWarnings = new ArrayNode(jsonNodeFactory);
+            doNodeValue.set("allowed_warnings", arrayWarnings);
+        }
+        allowedWarnings.forEach(arrayWarnings::add);
+    }
+
+    @Override
+    public String getKeyToFind() {
```

Comment:
I think some of these getters need @Input or similar annotations. I am getting gradle exceptions 
```
Property 'transformations.org.elasticsearch.gradle.test.rest.transform.headers.InjectHeaders$0.keyToFind' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0. See 
```

Reply:
I think these can be marked as `@Internal` since these are methods implemented by the transform class, and the class itself is tracked as an input. That is, these values only ever change if the transform implementation changes at which point gradle will consider the task out of date due to classpath changes.

Reply:
done: [`b60bf0c` (#69010)](https://github.com/elastic/elasticsearch/pull/69010/commits/b60bf0ca8b1e225abc205769a1ad7ea4b4b1b332)